src/syngen/ml/metrics/metrics_classes/metrics.py

```diff
@@ -897,25 +898,44 @@ def calculate_all(self, categ_columns: List[str], cont_columns: List[str]):
             self.original[col] = self.original[col].map(map_dict)
             self.synthetic[col] = self.synthetic[col].map(map_dict)
 
+        # Perform clustering of original dataset
+        # to get optimal number of clusters
+        original_for_clustering = self.original[
+            cont_columns + categ_columns
+            ].dropna()
+        original_transformed = self.__preprocess_data(original_for_clustering)
+
+        optimal_clust_num = self.__get_optimal_number_of_clusters(
+            original_transformed
+            )
+
         row_limit = min(len(self.original), len(self.synthetic))
+
         self.merged = (
             pd.concat(
                 [
-                    self.original[cont_columns + categ_columns].sample(row_limit),
-                    self.synthetic[cont_columns + categ_columns].sample(row_limit),
+                    self.original[cont_columns + categ_columns].sample(
+                        row_limit,
+                        random_state=10
```

Comment:
Do we still need random state here?

Reply:
Add pls #TODO reconsider necessity of random state

Reply:
Done